### PR TITLE
Refactor get all playlists with /playlists endpoint

### LINKED ISSUE #55: Refactor get all playlists with /playlists endpoint
Refactor current endpoint to return successful response as:

```[
  {
    "id": 1,
    "title": "Cleaning House",
    "songCount": 2,
    "songAvgRating": 27.5,
    "favorites": [
                    {
                      "id": 1,
                      "title": "We Will Rock You",
                      "artistName": "Queen"
                      "genre": "Rock",
                      "rating": 25
                    },
                    {
                      "id": 4,
                      "title": "Back In Black",
                      "artistName": "AC/DC"
                      "genre": "Rock",
                      "rating": 30
                    }
                  ],
    "createdAt": 2019-11-26T16:03:43+00:00,
    "updatedAt": 2019-11-26T16:03:43+00:00
  },
  {
    "id": 2,
    "title": "Running Mix",
    "songCount": 0,
    "songAvgRating": 0,
    "favorites": []
    "createdAt": 2019-11-26T16:03:43+00:00,
    "updatedAt": 2019-11-26T16:03:43+00:00
  },
]

### PR DESCRIPTION
**Pull request by zacisaacson**
## Type of change 

- [x] :beetle: Bug fix (non-breaking change which fixes an issue)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 

- Create happy path spec for GET /playlists
- Complete refactor of GET /playlists to include favorites
- Create helper method to find average rating
- Refactor GET /playlists/:id/favorites/:id using helper method

## How Has This Been Tested?

- [x] :white_square_button: Unit testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: helpers.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+const calculateSongAvgRating = (favorites) => {
+  if (favorites.length === 0) return 0
+  const totalRating = favorites.reduce((sum, favorite) => {
+    sum += favorite.rating
+    return sum
+  }, 0)
+  return totalRating / favorites.length
```
Comment (zacisaacson) [inline helpers.js]: Created a helper method for calculating song average, let me know what you think

File: routes/api/v1/playlists.js [hunk 2]
```diff
@@ -94,26 +114,18 @@ router.get('/:playlistId/favorites', (req, res) => {
...
-               .join('favorites', 'favorites.id', 'playlists_favorites.favorite_id')
-               .where('playlist_id', playlistId)
-               .avg('favorites.rating as songAvgRating')
-               .groupBy('playlists_favorites.playlist_id')
-               .then(songRating => {
-                 var rating = parseFloat(songRating[0].songAvgRating)
-                 songRating[0].songAvgRating = rating
-                 result[0].songAvgRating = songRating[0].songAvgRating
-                 res.status(200).json(result[0])
-               })
```
Comment (zacisaacson) [inline routes/api/v1/playlists.js]: Refactored this using the helper method

File: helpers.js (hunk 1 above)
Comment (lrs8810) [reply helpers.js]: Looks great! I think we might get feedback we should do this via a database query, but it looks cleaner than the one we had.

File: routes/api/v1/playlists.js (hunk 2 above)
Comment (lrs8810) [reply routes/api/v1/playlists.js]: Appreciate the consistency!

**Pull request merged**